sql/src/main/java/org/apache/druid/sql/calcite/aggregation/Aggregation.java

```diff
@@ -52,13 +52,13 @@ private Aggregation(
   )
   {
     this.aggregatorFactories = Preconditions.checkNotNull(aggregatorFactories, "aggregatorFactories");
-    this.postAggregator = postAggregator;
+    this.postAggregator = (postAggregator != null && postAggregator.getName() == null)? null :postAggregator;
 
     if (aggregatorFactories.isEmpty()) {
       Preconditions.checkArgument(postAggregator != null, "postAggregator must be present if there are no aggregators");
     }
-
-    if (postAggregator == null) {
+    //SSAGARE-Added postAggregator.getName in condition ass] we are passing postAgreegator with all variable as null
```

Comment:
you can remove SSAGARE, as the commit message and commit ID should contains the full details.